python/cudf/cudf/core/column/column.py

```diff
@@ -1017,7 +1017,9 @@ def distinct_count(
         return cpp_distinct_count(self, ignore_nulls=dropna)
 
     def astype(self, dtype: Dtype, **kwargs) -> ColumnBase:
-        if is_categorical_dtype(dtype):
+        if is_numerical_dtype(dtype):
```

Comment:
Numerical types being the most common [[citation needed]], and `is_numerical_dtype` now being quite fast, it makes sense to do this check first.

Reply:
Still definitely pro doing this, but I'm working on prototyping the (amortized) constant-time approach I suggested and I'll update you once that's done.